### PR TITLE
ES 6.8 Compatibility

### PR DESCRIPTION
**Pull request by willronchetti**
- Makes snovault compatible with ES 6.8, effectively deprecating ES 5

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 dist: trusty
 language: python
-sudo: true
+sudo:
```
Comment (netsettler) [inline .travis.yml:3]: Is this change semantic or just syntactic? (Does all of our travis checking normally run as a superuser or is this scoped somehow? I'm not actually worried about security concerns -- I assume this is a private instance only for the purpose of testing, but I wonder if we don't see some class of errors that our daemon users would get -- do they run as superusers, too?)

File: pyproject.toml [hunk 2]
```diff
@@ -45,8 +44,8 @@ awscli = "^1.15.42"
 #       so may no longer be needed. Something to investigate later. -kmp 20-Feb-2020
 "backports.statistics" = "0.1.0"
 boto3 = "^1.7.42"
-dcicutils = ">=0.37.0,<1"  # Needs LockoutManager introduced in 0.37.0
-elasticsearch_dsl = "^5.3.0"
+elasticsearch_dsl = "6.4.0"
+dcicutils = "0.42.0.0b0"
```
Comment (netsettler) [inline pyproject.toml:48]: You should pick a version one sub-minor version up from the most recent one (which will never be a .0 in the 4th position). n.n.n.0 will be seen the same as n.n.n and so I'm guessing the bug we're working around for versions like n.n.nb0 will likely be present on n.n.n.0b0, too. (I'd be on the lookout for problems related to this if you've deployed a version of this kind.)

File: snovault/tests/test_indexing.py [hunk 3]
```diff
@@ -426,6 +426,7 @@ def test_indexing_simple(app, testapp, indexer_testapp):
     uuid = res.json['@graph'][0]['uuid']
     res = indexer_testapp.post_json('/index', {'record': True})
     assert res.json['indexing_count'] == 1
+    time.sleep(1)  # failed on my machine without this -will 4-22-2020
```
Comment (netsettler) [inline snovault/tests/test_indexing.py:429]: How about using [`dcicutils.misc_utils.retrying`](https://github.com/4dn-dcic/utils/blob/master/dcicutils/misc_utils.py#L333) on the `testapp.get` (on the next line here) instead?

File: snovault/tests/test_views.py [hunk 4]
```diff
@@ -24,7 +24,8 @@
 
 def get_parameterized_names():
     """ Get all item types from schema names """
-    return [name.split('.')[0] for name in os.listdir(os.getcwd() + '/snovault/test_schemas')]
+    return [name.split('.')[0] for name in os.listdir(os.getcwd() + '/snovault/test_schemas')
```
Comment (netsettler) [inline snovault/tests/test_views.py:27]: Don't write tests against `cwd`. Make a variable at file toplevel based on `os.path.dirname(__file__)` or use `pkg_resources.resource_filename`. I prefer the former, but either is at least immune to working dir of the user, which should not be used.

File: .travis.yml (hunk 1 above)
Comment (willronchetti) [reply .travis.yml:3]: Seems like a syntactic change that was need awhile ago, perhaps it's not needed now. I think at one point we needed sudo to correctly spin up postgres on Travis but IRC sudo isn't actually needed to run the build.

File: pyproject.toml (hunk 2 above)
Comment (willronchetti) [reply pyproject.toml:48]: Once utils 1.0.0 is in this will be resolved.

File: snovault/elasticsearch/create_mapping.py [hunk 5]
```diff
@@ -610,7 +596,7 @@ def type_mapping(types, item_type, embed=True):
     type_info = types[item_type]
     schema = type_info.schema
     # use top_level parameter here for schema_mapping
-    mapping = schema_mapping('*', schema, True)
+    mapping = schema_mapping('*', schema, from_array=False)  # XXX: True seems to be a problem in es6?
```
Comment (netsettler) [inline snovault/elasticsearch/create_mapping.py:599]: Can you add a longer comment explaining the issue here? What is lost by making this change?

File: snovault/elasticsearch/create_mapping.py [hunk 6]
```diff
@@ -642,6 +629,13 @@ def type_mapping(types, item_type, embed=True):
                                schema['properties'][curr_e]['type'] == 'array')
             if map_with_nested:
                 curr_m['type'] = "nested"
+
+
+    # Copy text fields to 'full_text' so we can still do _all searches
+    properties = schema['properties']
+    for _, sub_mapping in properties.items():  # TODO: filter based on field_name is probably a good idea
```
Comment (netsettler) [inline snovault/elasticsearch/create_mapping.py:636]: This comment could be a little more elaborate, perhaps citing an example of what is not being filtered that should or could.

File: snovault/tests/test_create_mapping.py [hunk 7]
```diff
@@ -95,3 +94,42 @@ def test_update_mapping_by_embed(registry):
...
+
+
+# just testing order, nothing bigger than that here.
+ORDER = ['testing_mixins', 'embedding_test', 'nested_embedding_container', 'nested_object_link_target',
+         'testing_download', 'testing_link_source_sno', 'testing_link_aggregate_sno', 'testing_link_target_sno',
+         'testing_post_put_patch_sno', 'testing_server_default', 'testing_dependencies',
+         'testing_link_target_elastic_search']
+
+
+@pytest.mark.parametrize('item_type', ORDER)
```
Comment (netsettler) [inline snovault/tests/test_create_mapping.py:106]: Is this really relying on the tests to run in the indicated order? Isn't the variable ORDER used elsewhere in some magic way? Is this capitalizing or influencing that magic? If not, I might all this ITEM_TYPE_TESTING_ORDER.

But also, why is it an ordering? Would these not work well if done in another order? The comment above the assignment seems to play down the order but I feel like there's something unspoken here. Maybe I'm thinking you should just have a loop within the test. Could this all be a single test where you're pulling them all into the same test? That would make it clear they all rely on going in a certain order in a way I feel is being obfuscated (and left at risk) here.

Comment (willronchetti) [reply snovault/tests/test_create_mapping.py:105]: Yeah, another holdover that isn't that meaningful here. I will change it.

File: snovault/elasticsearch/indexer_utils.py [hunk 8]
```diff
@@ -45,8 +45,7 @@ def find_uuids_for_indexing(registry, updated, find_index=None):
                         'should': [
                             {
                                 'terms': {
-                                    'linked_uuids_embedded.uuid': list(updated),
-                                    '_cache': False,
```
[the remark this answers is not in the document]
Comment (willronchetti) [reply snovault/elasticsearch/indexer_utils.py:49]: Good catch - as it turns out, this never actually did anything: https://www.elastic.co/guide/en/elasticsearch/reference/6.8/breaking-changes-6.0.html#breaking_60_search_changes

**Pull request merged**